py3status/modules/coin_market.py

```diff
@@ -0,0 +1,152 @@
+        (default ['btc'])
+    request_timeout: time to wait for a response, in seconds (default 5)
+
+Format placeholder:
+    {format_coin} format for cryptocurrency coins
+
+format_coin placeholders:
+    {24h_volume_usd}      eg 1435150000.0
+    {available_supply}    eg 16404825.0
+    {id}                  eg bitcoin
+    {last_updated}        eg 1498135152
+    {market_cap_usd}      eg 44119956596.0
+    {name}                eg Bitcoin
+    {percent_change_1h}   eg -0.17%
+    {percent_change_24h}  eg -1.93%
+    {percent_change_7d}   eg +14.73%
+    {price_btc}           eg 1.0
+    {price_usd}           eg 2689.45
+    {rank}                eg 1
+    {symbol}              eg BTC
+    {total_supply}        eg 16404825.0
+
+    Placeholders are retrieved directly from the URL.
+    The list was harvested only once and should not represent a full list.
+
+    To view placeholders in different currency, replace `_usd` with a valid option
+    (eg '{price_usd}' to '{price_xxx}'). Valid options are: AUD, BRL, CAD, CHF,
+    CNY, EUR, GBP, HKD, IDR, INR, JPY, KRW, MXN, RUB, otherwise USD.
+    Valid options must be written in lowercase.
+
+Examples:
+```
+# view coins in GBP
+coin_market {
+    format_coin = '[\?color=degraded {name}] £{price_gbp:.2f} {percent_change_24h}'
+}
+# change name color
+coin_market {
+    color_degraded = '#3DAEE9'
+}
+```
+
+@author x86kernel, lasers
+
+SAMPLE OUTPUT
+[
+    {'color': u'#FFFF00', 'full_text': u'Bitcoin'},
+    {'full_text': u' $2735.77 '},
+    {'color': '#00FF00', 'full_text': u'+2.27%'},
+]
+
+losers
+[
+    {'color': u'#FFFF00', 'full_text': u'Bitcoin'},
+    {'full_text': u' $2701.70 '},
+    {'color': '#FF0000', 'full_text': u'-0.42%'},
+]
+"""
+
+
+class Py3status:
+    cache_timeout = 600
+    format = '{format_coin}'
+    format_coin = '[\?color=degraded {name}] ${price_usd:.2f} {percent_change_24h}'
+    format_separator = ' '
+    markets = ['btc']
+    request_timeout = 5
+
+    def post_config_hook(self):
+        self.first = True
+        if not isinstance(self.markets, list):
+            self.markets = self.markets.split(',')
+        self.markets = [x.upper().strip() for x in self.markets]
+        self.url = 'https://api.coinmarketcap.com/v1/ticker/?convert=%s'
+        for item in self.py3.get_placeholders_list(self.format_coin):
+            if (('price_' in item and 'price_btc' not in item) or
+                    '24h_volume_' in item or 'market_cap_' in item) and 'usd' not in item:
+                self.url = self.url % (item.split('_')[-1])
+                break
+
+        self.coin_data = self._get_coin_data()
+        limit = 0
+        for market_id in self.markets:
+            index = next((i for (
+                i, d) in enumerate(self.coin_data) if d['symbol'] == market_id), -1)
+            if index > limit:
+                limit = index
+        self.url = self.url + '&limit=%s' % (limit + 1)
```

Comment:
Maybe just do a fresh call of them all if you can't find all the required currencies (I'm going to assume that the order changes rarely.  So move that code out of post config into it's own function etc